crates/bevy_core_pipeline/src/core_2d/main_pass_2d_node.rs

```diff
@@ -57,37 +59,60 @@ impl Node for MainPass2dNode {
+            let _main_pass_2d = info_span!("main_pass_2d").entered();
+            let pass_descriptor = RenderPassDescriptor {
+                label: Some("main_pass_2d"),
+                color_attachments: &[target.get_color_attachment(Operations {
+                    load: match camera_2d.clear_color {
+                        ClearColorConfig::Default => {
+                            LoadOp::Clear(world.resource::<ClearColor>().0.into())
+                        }
+                        ClearColorConfig::Custom(color) => LoadOp::Clear(color.into()),
+                        ClearColorConfig::None => LoadOp::Load,
+                    },
+                    store: true,
+                })],
+                depth_stencil_attachment: None,
+            };
 
-        let pass_descriptor = RenderPassDescriptor {
-            label: Some("main_pass_2d"),
-            color_attachments: &[target.get_color_attachment(Operations {
-                load: match camera_2d.clear_color {
-                    ClearColorConfig::Default => {
-                        LoadOp::Clear(world.resource::<ClearColor>().0.into())
-                    }
-                    ClearColorConfig::Custom(color) => LoadOp::Clear(color.into()),
-                    ClearColorConfig::None => LoadOp::Load,
-                },
-                store: true,
-            })],
-            depth_stencil_attachment: None,
-        };
-
-        let draw_functions = world.resource::<DrawFunctions<Transparent2d>>();
+            let draw_functions = world.resource::<DrawFunctions<Transparent2d>>();
 
-        let render_pass = render_context
-            .command_encoder
-            .begin_render_pass(&pass_descriptor);
+            let render_pass = render_context
+                .command_encoder
+                .begin_render_pass(&pass_descriptor);
 
-        let mut draw_functions = draw_functions.write();
-        let mut tracked_pass = TrackedRenderPass::new(render_pass);
-        if let Some(viewport) = camera.viewport.as_ref() {
-            tracked_pass.set_camera_viewport(viewport);
+            let mut draw_functions = draw_functions.write();
+            let mut tracked_pass = TrackedRenderPass::new(render_pass);
+            if let Some(viewport) = camera.viewport.as_ref() {
+                tracked_pass.set_camera_viewport(viewport);
+            }
+            for item in &transparent_phase.items {
+                let draw_function = draw_functions.get_mut(item.draw_function).unwrap();
+                draw_function.draw(world, &mut tracked_pass, view_entity, item);
+            }
         }
-        for item in &transparent_phase.items {
-            let draw_function = draw_functions.get_mut(item.draw_function).unwrap();
-            draw_function.draw(world, &mut tracked_pass, view_entity, item);
+
+        // If there was a camera with a viewport, finish with a renderpass without a custom viewport in webgl
+        #[cfg(feature = "webgl")]
```

Comment:
I think its worth directly calling out the behavior quirks of WebGL explicitly, to better justify this code to future readers.
On that note, I'm still a little unclear about what the behavior actually is. Is it just "the next renderpass will still have the viewport set, but every pass after that will be fine"?

Should we open a wgpu bug?

Reply:
This is what it looks like without that extra render pass
<img width="1456" alt="Screenshot 2022-06-09 at 01 23 06" src="https://user-images.githubusercontent.com/8672791/172733577-a5f97fe0-8827-4437-86b0-fbc03d5864ed.png">

Reply:
I tried making the comment more explicit. I don't know if it's a wgpu bug or how WebGL2 works, but I found a lot of people trying to reset viewport when using OpenGL

Reply:
Gotcha. Its a bit unfortunate because this means users developing things like custom post-processing effects will need to manually hard-code the reset if they want to support wasm viewports. But thats not something we can reasonably fix in this pr.